google/cloud/bigtable/internal/async_list_instances.h

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_INTERNAL_ASYNC_LIST_INSTANCES_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_INTERNAL_ASYNC_LIST_INSTANCES_H_
+
+#include "google/bigtable/admin/v2/bigtable_instance_admin.grpc.pb.h"
+#include "google/cloud/bigtable/completion_queue.h"
+#include "google/cloud/bigtable/instance_admin_client.h"
+#include "google/cloud/bigtable/internal/async_retry_multi_page.h"
+#include "google/cloud/bigtable/metadata_update_policy.h"
+#include "google/cloud/internal/make_unique.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+struct InstanceList {
```

Comment:
If this is a data structure that the application developers are supposed to use we should move its definition to a file in `google/cloud/bigtable/`, otherwise it is not included in the Doxygen documentation.  We also need to document it.

Reply:
I need to fix for List Cluster as well, so later I will do that during implementation of ListAppProfiles

Reply:
You are correct, I filed a bug (#1573) for that.